src/main/java/TaskList.java

```diff
@@ -0,0 +1,223 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+public class TaskList {
+
+    private ArrayList<Task> taskList;
+    private int totalTask;
+
+    public TaskList() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    public TaskList(ArrayList<Task> myTasks, int totalTask) {
+        this.taskList = myTasks;
+        this.totalTask = totalTask;
+    }
+
+    public TaskList(ArrayList<String> myTasks) {
+        this.taskList = initialiseList(myTasks);
+        this.totalTask = getSize();
+    }
+
+    public TaskResult executeOperation(TaskAction taskAction) throws DukeException {
+        Task relatedTask = taskAction.getRelatedTask();
+        int relatedTaskNumber = taskAction.getRelatedTaskNumber();
+        String keyword = taskAction.getKeyword();
+        String actionType = taskAction.getActionType();
+
+        TaskResult taskResult = new TaskResult();
+        switch (actionType){
+
+        case "add":
+            addTask(relatedTask);
+            taskResult = new TaskResult(relatedTask, "add");
+            break;
+
+        case "complete":
+            relatedTask = markTaskAsDone(relatedTaskNumber);
+            taskResult = new TaskResult(relatedTask, "complete");
+            break;
+
```

Comment:
Perhaps it could be nicer to read without the extra line-breaks between the cases?